lib/storage.ts

```diff
@@ -14,3 +14,17 @@ export async function loadCharacter(characterId: string): Promise<CharacterType>
 export async function saveCharacter(character: CharacterType): Promise<void> {
   await kv.json.set(`character:${character.characterId}`, '$', character);
 }
+
+/** Load the Character ID mapped to the given Agent ID from KV. */
+export async function loadCharacterIdByAgentId(agentId: string): Promise<string> {
+  const characterId = await kv.json.get(`agent-character:${agentId}`);
+  if (characterId === null) {
+    throw new Error(`Character ID not found for Agent ${agentId}`);
+  }
+  return characterId;
+}
+
+/** Store the mapping of Agent ID to Character ID to KV. */
+export async function saveAgentCharacterMapping(agentId: string, characterId: string): Promise<void> {
```

Comment:
Instead please update saveCharacter to write the additional mapping by agentID. I would just call this `agent:${agentId}` and leave off the `-character` part of the key.